File: Bank1/BeckTDD/Money/Sum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TDDMoney
{
    public class Sum : Expression
    {
        public Expression AugEnd;
        public Expression AddEnd;

        public Sum(Expression augend, Expression addend)
        {
            this.AugEnd = augend;
            this.AddEnd = addend;
        }

        public Expression Plus(Expression addend)
        {
            return new Sum(this, addend);
        }

        public Expression Times(int multiplier)
        {
            return new Sum(AugEnd.Times(multiplier), AddEnd.Times(multiplier));
        }

        public Money Reduce(Bank bank, string to)
        {
            int amount = AugEnd.Reduce(bank, to).amount + AddEnd.Reduce(bank, to).amount;
            return new Money(amount, to);
        }
    }
}

File: Bank1/src/sum.js
class sum{
	constructor(Expression augend,Expression addend){
	this.augend = augend;
	this.addend = addend;
	}
	
    get Moneyreduce(bank,to ) {
        int amount = augend.reduce(bank, to).amount + addend.reduce(bank, to).amount;
        return new Money(amount, to);
    }/*doubt*/

    @Override
	var expression plus(addend) {
	return new Sum(this, addend);  /* not sure of how to override in JS*/
	}

    var times(multiplier) {
        return new Sum(augend.times(multiplier), addend.times(multiplier));
    }
}